### PR TITLE
llvm: add libc++abi options

### PR DESCRIPTION
**Pull request by wangpeiwen**
fix some problem on linux

- [ ] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by maxim-belkin** [comment]
@wangpeiwen could you please share gist-logs?

**Comment by mikeandmore** [comment]
I believe the major reason libcxxabi is removed (in https://github.com/Homebrew/homebrew-core/pull/4170) is because of https://github.com/Homebrew/homebrew-core/issues/4081 in homebrew upstream.

**Comment by wangpeiwen** [comment]
https://gist.github.com/anonymous/e4bcaba68db0adcefc8e1bd38cb83c34
this is gist logs for build llvm failed when with-libc++
libc++abi is necessary on linux so I add a option to do it.
and install-xcode-toolchain make `make install` failed on linux because there is no xcode
libpython.dylib is same on linux

**Comment by maxim-belkin** [comment]
Could you please rebase this PR againts the latest llvm formula?
```bash
git checkout master
git pull origin
git checkout <branch-name>
git rebase master
```

**Comment by wangpeiwen** [comment]
i have done it

File: Formula/llvm.rb [hunk 1]
```diff
@@ -47,6 +47,11 @@ class Llvm < Formula
       sha256 "25e615e428f60e651ed09ffd79e563864e3f4bc69a9e93ee41505c419d1a7461"
     end
 
+    resource "libcxxabi" do
+      url "http://llvm.org/releases/3.9.0/libcxxabi-3.9.0.src.tar.xz"
```
Comment (rwhogg) [inline Formula/llvm.rb]: I think this should be changed to 3.9.1 (sorry, LLVM upgraded when I merged Homebrew/core).

Comment (wangpeiwen) [inline Formula/llvm.rb]: have changed to 3.9.1

**Comment by maxim-belkin** [comment]
I've merged the fix for Python library name in #1533. Thanks for noticing that one!
Could you please rebase this PR?

**Comment by wangpeiwen** [comment]
done

**Comment by sjackman** [comment]
Could you please confirm that `llvm` strictly requires `libc++abi` on Linux? I'm not up to date with the status of `llvm` on Linux.

**Comment by wangpeiwen** [comment]
`libc++abi ` seems to be not necessary for `llvm` but for `libc++` on linux.
so i add a option for it

**Comment by maxim-belkin** [comment]
> `libc++abi ` seems to be not necessary for `llvm` but for `libc++` on linux.
so i add a option for it

Does it it mean that there should be no option to install `libcxxabi` on its own and we should install it whenever we install `libcxx`?

**Comment by wangpeiwen** [comment]
but i'm not sure `libcxxabi` is need by `libcxx` on all linux
so I choose to add a option
maybe we can add a recommand for it?

**Comment by sjackman** [comment]
What's your use case for using LLVM's `libcxx` rather than GNU's `libstdc++`?

**Comment by wangpeiwen** [comment]
I just want to test with diffrent c++ standard library

**Comment by sjackman** [comment]
> but i'm not sure libcxxabi is need by libcxx on ALL linux

Does anyone know the answer to this question? Is it possible to build `libcxx` without `libcxxabi`?

**Comment by mikeandmore** [comment]
@sjackman @wangpeiwen Yes. I can confirm libc++ needs libc++abi. See https://github.com/Linuxbrew/homebrew-core/issues/1041

libc++ in general needs a c++ runtime library, and given that libc++ implements a majority of std:: stuff, I don't think it can ever link with libstdc++ from gcc. I'd say it's a wise idea to let libc++ depend on libc++abi.

**Comment by maxim-belkin** [comment]
@wangpeiwen please see my comments [here](https://github.com/Linuxbrew/homebrew-core/pull/1413/files). They are easy to address.

**Comment by wangpeiwen** [comment]
@maxim-belkin i'm sorry, but i can't see any comments there

File: Formula/llvm.rb [hunk 2]
```diff
@@ -131,6 +140,7 @@ class Llvm < Formula
     option "with-libcxx", "Build libc++ standard library"
   end
   option "with-toolchain", "Build with Toolchain to facilitate overriding system compiler"
+  option "with-libcxxabi", "Build with libc++abi"
```
Comment (sjackman) [inline Formula/llvm.rb]: Remove this option.

File: Formula/llvm.rb [hunk 3]
```diff
@@ -195,6 +205,7 @@ def install
     (buildpath/"tools/clang/tools/extra").install resource("clang-extra-tools")
     (buildpath/"projects/openmp").install resource("openmp")
     (buildpath/"projects/libcxx").install resource("libcxx") if build_libcxx?
+    (buildpath/"projects/libcxxabi").install resource("libcxxabi") if build.with? "libcxxabi"
```
Comment (sjackman) [inline Formula/llvm.rb]: Change to
```ruby
(buildpath/"projects/libcxxabi").install resource("libcxxabi") if build_libcxx? && !OS.mac?
```

File: Formula/llvm.rb [hunk 4]
```diff
@@ -255,6 +266,7 @@ def install
     end
 
     args << "-DLLVM_ENABLE_LIBCXX=ON" if build_libcxx?
+    args << "-DLLVM_ENABLE_LIBCXXABI=ON" if build.with? "libcxxabi"
```
Comment (sjackman) [inline Formula/llvm.rb]: Change to
```ruby
args << "-DLLVM_ENABLE_LIBCXXABI=ON" if build_libcxx? && !OS.mac?
```

**Comment by wangpeiwen** [comment]
done

**Comment by sjackman** [comment]
Merged. Thanks for your contribution to Linuxbrew!

**Pull request closed without merging**